src/_pytest/hookspec.py

```diff
@@ -22,17 +23,27 @@ def pytest_addhooks(pluginmanager):
     """
 
 
-@hookspec(historic=True)
+@hookspec(
+    historic=True,
+    warn_on_impl=RemovedInPytest4Warning(
+        "pytest_namespace is deprecated and will be removed soon"
+    ),
+)
 def pytest_namespace():
     """
-    (**Deprecated**) this hook causes direct monkeypatching on pytest, its use is strongly discouraged
     return dict of name->object to be made globally available in
     the pytest namespace.
 
     This hook is called at plugin registration time.
 
     .. note::
         This hook is incompatible with ``hookwrapper=True``.
+
+    .. warning::
+        This hook has been **deprecated** and will be removed in pytest 4.0.
+
+        Plugins who need this feature are suggested
```

Comment:
i propose to align this somewhere along
```
Plugins whose users depend on the current name-space functionality should prepare to migrate to a namespace they actually own.

To support the migration its suggested to trigger `DeprecationWarnings` for objects they put into the pytest namespace.

An stopgap measure to avoid the warning is to monkeypatch the `pytest` module.
but just as the `pytest_namespace` hook this should be seen as a temporary measure to be removed.

```
simply to avoid accidentally declaring monkeypatching ´pytest` fine

Reply:
Sure will do

Reply:
Done